server/rootless.go

```diff
@@ -17,7 +21,41 @@ func hasNetworkNamespace(config *rspec.Spec) bool {
 }
 
 func makeOCIConfigurationRootless(g *generate.Generator) {
-	g.Config.Linux.Resources = nil
+	// Resource limitations requires cgroup v2 delegation (https://rootlesscontaine.rs/getting-started/common/cgroup2/).
+	if g.Config.Linux.Resources != nil {
+		// cannot control device eBPF with rootless
+		g.Config.Linux.Resources.Devices = nil
+		v2Controllers := getAvailableV2Controllers()
+		if _, ok := v2Controllers["memory"]; !ok && g.Config.Linux.Resources.Memory != nil {
+			logrus.Warn("rootless: cgroup v2 memory controller is not delegated. Discarding memory limit.")
+			g.Config.Linux.Resources.Memory = nil
+		}
+		if _, ok := v2Controllers["cpu"]; !ok && g.Config.Linux.Resources.CPU != nil {
```

Comment:
nit: any reason to not unconditionally set ...CPU = nil if we don't have the controller? i.e. why are we checking if the field is not nil before setting it to be nil?

Reply:
Because when CPU is already nil, no need to print the warning. (But I haven't checked whether CPU can be nil)